Feat: "Add login, signup"

jeongminy opened the pull request:
Feat: "Add login, signup"

회원가입, 로그인 개발

Ref: #4

jeongminy committed:
Feat: "Add login, signup"

회원가입, 로그인 개발

Ref: #4

9nh5 reviewed (approved):
확인했어요!

HwangSeungHyeon commented on src/main/kotlin/com/teamsparta/bunnies/domain/exception/GlobalExceptionHandler.kt:
```diff
@@ -0,0 +1,7 @@
+import org.springframework.web.bind.annotation.RestControllerAdvice
+
+@RestControllerAdvice
+class GlobalExceptionHandler {
```
ModelNotFoundException과 INvalidCredentialException이 등록이 안되어있습니당

HwangSeungHyeon commented on src/main/resources/application.yml:
```diff
@@ -1 +1,5 @@
+auth:
+  jwt:
+    issuer: team.sparta.com
+    secret: PO4c8z41Hia5gJG3oeuFJMRYBB4Ws4aZ
```
secret 같은 경우에는 팀원끼리만 공유를 하고, 원래는 github에 올리지 않는 게 좋습니다.
secret이 유출될 경우 계정이 털린 것과 마찬가지인 상태가 됩니다.

HwangSeungHyeon commented on src/main/kotlin/com/teamsparta/bunnies/infra/security/SecurityConfig.kt:
```diff
@@ -0,0 +1,46 @@
+            .authorizeHttpRequests {
+                it.requestMatchers(
+                    "/api/login",
+                    "/api/signup",
```
signup과 login같은 경우에는 익명의 사용자만 접근하는 편이 더 좋을 것 같습니다.
로그인된 상태에서 또 로그인 페이지에 접근하지 않아도 된다고 생각합니다.

HwangSeungHyeon reviewed (commented):
코드 리뷰한 거 확인해주세요

YunmiC commented on src/main/kotlin/com/teamsparta/bunnies/domain/user/dto/response/LoginResponse.kt:
```diff
@@ -0,0 +1,5 @@
+
+class LoginResponse (
+    val accessToken: String
+)
```
리뷰연습

YunmiC reviewed (commented):
승인

jeongminy committed:
Merge remote-tracking branch 'origin/develop' into feature/user

# Conflicts:
#	build.gradle.kts
#	src/main/kotlin/com/teamsparta/bunnies/infra/swagger/SwaggerConfig.kt
#	src/main/resources/application.yml

jeongminy committed:
Feat: "Add login, signup"

SwaggerConfig 임포트 수정

Ref: #4

jeongminy committed:
Feat: "Add login, signup"

ErrorResponse -> ErrorResponseDto 로 수정

Ref: #4

jeongminy committed:
Feat: "Add login, signup"

Request -> RequestDto 로 이름 변경
Response -> ResponseDto 로 이름 변경

Ref: #4

jeongminy committed:
Feat: "Add login, signup"

Request -> RequestDto 로 이름 변경
Response -> ResponseDto 로 이름 변경

Ref: #4

jeongminy committed:
Feat: "Add login, signup"

Entity -> EntityDto 로 이름 변경

Ref: #4

jeongminy committed:
Feat: "Add login, signup"

UserRole.USER -> UserRole 변경

Ref: #4

jeongminy committed:
Feat: "Add login, signup"

GlobalExceptionHandler 에 handleInvalidCredentialException 추가
GlobalExceptionHandler 에 handleModelNotFoundException 추가

Ref: #4

jeongminy committed:
Feat: "Add login, signup"

IllegalStateException 의 message 수정

Ref: #4

jeongminy committed:
Feat: "Add login, signup"

Entity -> EntitoDto 로 이름 변경

Ref: #4

jeongminy committed:
Feat: "Add login, signup"

jwt.yml 생성 후 JwtPlugin 코드 변경

Ref: #4

jeongminy committed:
Feat: "Add login, signup"

익명의 사용자가 post를 조회 할 수 있도록 filterChain 로직 변경

Ref: #4

jeongminy committed:
Feat: "Add login, signup"

익명의 사용자가 post를 조회 할 수 있도록 filterChain 로직 변경

Ref: #4

jeongminy committed:
Feat: "Add login, signup"

Postgresql DB 연결

Ref: #4

jeongminy committed:
Feat: "Add login, signup"

Info 변경

Ref: #4

jeongminy committed:
Feat: "Add login, signup"

swawgger -> swagger 오타 수정

Ref: #4

9nh5 reviewed (approved):
긋긋
